editorextensions/codebeamer-custom-ui/modal/src/pages/import/components/importer/Importer.tsx

```diff
@@ -44,22 +45,16 @@ export default function Importer(props: {
 	//* applies all currently active filters by using the stored cbqlString,
 	//* then further filters out only the selected items (or takes all of 'em)
 
-	// const { data, error, isLoading } = useGetItemsQuery({
-	// 	page: DEFAULT_RESULT_PAGE,
-	// 	pageSize: MAX_ITEMS_PER_IMPORT,
-	// 	queryString: `${getMainQueryString()}${
-	// 		importedItems.length
-	// 			? ' AND item.id NOT IN (' +
-	// 			  importedItems.map((i) => i.itemId) +
-	// 			  ')'
-	// 			: ''
-	// 	}`,
-	// });
-
 	const { data, error, isLoading } = useGetItemsQuery({
 		page: DEFAULT_RESULT_PAGE,
 		pageSize: MAX_ITEMS_PER_IMPORT,
-		queryString: getMainQueryString(),
+		queryString: `${getMainQueryString()}${
+			importedItems.length
+				? ' AND item.id NOT IN (' +
+				  importedItems.map((i: { itemId: number }) => i.itemId) +
+				  ')'
+				: ''
+		}`,
```

Comment:
We already have a function that is supposed to do all the query string building. `getMainQueryString` - Why put this additional logic outside of it?

Reply:
this was actually your code that I uncommented but I changed it now to be inside of the function itself.